core/templates/dev/head/components/ck-editor-helpers/ck-editor-5-rte.directive.ts

```diff
@@ -88,17 +88,76 @@ angular.module('oppia').directive('ckEditor5Rte', [
           scope.uiConfig().startupFocusEnabled !== undefined) {
           startupFocusEnabled = scope.uiConfig().startupFocusEnabled;
         }
+        // A RegExp for matching rich text components.
+        var componentRegExp = (
+          /(<(oppia-noninteractive-(.+?))\b[^>]*>)[\s\S]*?<\/\2>/g
+        );
+        var wrapComponents = function(html) {
+          if (html === undefined) {
+            return html;
+          }
+          // This wraps the custom plugin data currently, we are using only
+          // default tool bar.
+          return html.replace(componentRegExp, function(match, p1, p2, p3) {
+            if (RteHelperService.isInlineComponent(p3)) {
+              return '<span type="oppia-noninteractive-' + p3 + '">' +
+                    match + '</span>';
+            } else {
+              return '<div type="oppia-noninteractive-' + p3 + '"' +
+                     'class="oppia-rte-component-container">' + match +
+                     '</div>';
+            }
+          });
+        };
         // CkEditor5 is initalized to the editable element which is passed
         // through the create api. el[0] is the ck-editor-5-rte and
         // el[0].children[0].children[1] is the contenteditable div which
         // is defined in the template above.
-        var ck = ClassicEditor.create(
-          <HTMLElement>(elem[0].children[0].children[1]));
+        ClassicEditor.create(<HTMLElement>(
+          elem[0].children[0].children[1])).then(ck => {
+          // This sets the CKEditor-5 data from the model view.
+          ck.setData(wrapComponents(ngModel.$viewValue));
 
-        // A RegExp for matching rich text components.
-        var componentRegExp = (
-          /(<(oppia-noninteractive-(.+?))\b[^>]*>)[\s\S]*?<\/\2>/g
-        );
+          // This tracks the data change in CKEDitor-5 and updates the view if
+          // any data change is observed.
+          ck.model.document.on('change', function() {
+            var elt = $('<div>' + ck.getData() + '</div>');
+            // The data from CKEditor5 will be in CK-format it needs to
+            // be refined for need html format.
+            // read more here: https://github.com/oppia/oppia/issues/5400.
+            var tagName;
```

Comment:
You can just drop this line and define "var tagName" on line 132 right?

Reply:
Done PTAL!